### PR TITLE
Add Connect ads to tsh login and tsh proxy db

### LINKED ISSUE #28991: Advertise Connect in tsh
Advertise Connect by linking to [its documentation](https://goteleport.com/docs/connect-your-client/teleport-connect/) when running the following commands:

* tsh login
   * `https://goteleport.com/docs/connect-your-client/teleport-connect/`
* tsh proxy db
   * `https://goteleport.com/docs/connect-your-client/teleport-connect/#connecting-to-a-database`

TODO:

* [ ] Make sure to test with MOTD enabled and a cluster alert, just to see what it looks like with all 3 messages printed.
* [ ] Coordinate with Ben to set up posthog URL tracking for this.

### PR DESCRIPTION
**Pull request by ravicious**
Closes #28991.

The motivation behind this is described in #28991.

I've tried to always show the ad before any more important messages so that the most important message is at the end of the output where the user is most likely to see it (that is my assumption). In `tsh login`, the ad is shown before the alerts. In `tsh proxy db`, the ad is shown before the instructions on how to use the newly set up proxy.

Copy suggestions are welcome. I'm particularly not sure about the `tsh proxy db`. Adding "Did you know?" there felt unnatural as, unlike in `tsh login`, the ad there is not completely random but rather directly related to what you're attempting to do.

This PR will be followed by some updates to the Connect page in docs. I want to add a section at the top describing why someone would want to use Connect over tsh.

---

Ad in `tsh login`:

```
Did you know? Teleport Connect offers the power of tsh in a desktop app.
Learn more at https://goteleport.com/docs/connect-your-client/teleport-connect/
```

Ad in `tsh proxy db`:

```
Teleport Connect is a desktop app that can manage database proxies for you.
Learn more at https://goteleport.com/docs/connect-your-client/teleport-connect/#connecting-to-a-database
```

Examples:

<details>
<summary>tsh login</summary>

local login with no motd and alerts

```
~teleport(master) ✗: tsh login --proxy=enterprise-local:3030
Enter password for Teleport user rav:
> Profile URL:        https://enterprise-local:3030
  Logged in as:       rav
  Cluster:            enterprise-local
  Roles:              access, auditor, connect-my-computer-rav, db, editor, kube-access, sbar-admins
  Logins:             root, rav
  Kubernetes:         enabled
  Kubernetes users:   minikube
  Kubernetes groups:  viewers
  Valid until:        2023-07-12 22:55:46 +0200 CEST [valid for 10h10m0s]
  Extensions:         login-ip, permit-agent-forwarding, permit-port-forwarding, permit-pty, private-key-policy

  Profile URL:        https://teleport-local.dev:3090
  Logged in as:       rav
  Cluster:            teleport-local
  Roles:              access, connect-my-computer-rav, editor, role-with-label-expressions
  Logins:             root, rav
  Kubernetes:         enabled
  Kubernetes users:   minikube
  Kubernetes groups:  viewers
  Valid until:        2023-07-12 20:44:13 +0200 CEST [valid for 7h59m0s]
  Extensions:         login-ip, permit-agent-forwarding, permit-port-forwarding, permit-pty, private-key-policy

Did you know? Teleport Connect offers the power of tsh in a desktop app.
Learn more at https://goteleport.com/docs/connect-your-client/teleport-connect/

~teleport(master) ✗:
```

local login with motd and three alerts

```
~teleport(master) ✗: tsh login --proxy=teleport-local.dev:3090
=====================================================
Welcome to Zion's Teleport Cluster.

Remember, you are no longer in the Matrix:

1. All actions performed here are free from the Machines' surveillance.
2. Unauthorized access will face the wrath of the Oracle.
3. System maintenance window: When the Matrix resets.

Remember: The Matrix is not real. Our fight for freedom is.

For any issues, contact the Oracle or Morpheus.

=====================================================

Press [ENTER] to continue.
Enter password for Teleport user rav:
> Profile URL:        https://teleport-local.dev:3090
  Logged in as:       rav
  Cluster:            teleport-local
  Roles:              access, connect-my-computer-rav, editor, role-with-label-expressions
  Logins:             root, rav
  Kubernetes:         enabled
  Kubernetes users:   minikube
  Kubernetes groups:  viewers
  Valid until:        2023-07-12 20:44:13 +0200 CEST [valid for 8h0m0s]
  Extensions:         login-ip, permit-agent-forwarding, permit-port-forwarding, permit-pty, private-key-policy

Did you know? Teleport Connect offers the power of tsh in a desktop app.
Learn more at https://goteleport.com/docs/connect-your-client/teleport-connect/

All systems down. Contact IT Emergency: it-emergency@zion.com immediately.

Unusual system activity detected. Please save work often and report any anomalies.

Minor network issues detected. No immediate action required, but expect possible slowdowns.

~teleport(master) ✗:
```

SSO with an upgrade alert:

```
~teleport(master) ✗: tsh login --proxy=teleport-13-ent.asteroid.earth --auth=github
If browser window does not open automatically, open it by clicking on the link:
 http://127.0.0.1:61653/35ec203f-68b6-4f76-856f-795d674cd5b6
> Profile URL:        https://teleport-13-ent.asteroid.earth:443
  Logged in as:       ravicious
  Cluster:            teleport-13-ent.asteroid.earth
  Roles:              access
  Logins:             ravicious, ubuntu
  Kubernetes:         enabled
  Kubernetes groups:  system:masters
  Valid until:        2023-07-13 00:49:44 +0200 CEST [valid for 12h0m0s]
  Extensions:         login-ip, permit-agent-forwarding, permit-port-forwarding, permit-pty, private-key-policy

  Profile URL:        https://enterprise-local:3030
  Logged in as:       rav
  Cluster:            enterprise-local
  Roles:              access, auditor, connect-my-computer-rav, db, editor, kube-access, sbar-admins
  Logins:             root, rav
  Kubernetes:         enabled
  Kubernetes users:   minikube
  Kubernetes groups:  viewers
  Valid until:        2023-07-12 22:55:46 +0200 CEST [valid for 10h6m0s]
  Extensions:         login-ip, permit-agent-forwarding, permit-port-forwarding, permit-pty, private-key-policy

  Profile URL:        https://teleport-local.dev:3090
  Logged in as:       rav
  Cluster:            teleport-local
  Roles:              access, connect-my-computer-rav, editor, role-with-label-expressions
  Logins:             root, rav
  Kubernetes:         enabled
  Kubernetes users:   minikube
  Kubernetes groups:  viewers
  Valid until:        2023-07-12 20:44:13 +0200 CEST [valid for 7h54m0s]
  Extensions:         login-ip, permit-agent-forwarding, permit-port-forwarding, permit-pty, private-key-policy

Did you know? Teleport Connect offers the power of tsh in a desktop app.
Learn more at https://goteleport.com/docs/connect-your-client/teleport-connect/

Teleport v13.2.0 is now available, please consider upgrading your Cluster.

~teleport(master) ✗:
```

</details>

<details>
<summary>tsh proxy db</summary>

tunnel with random port

```
teleport(master) ✗: tsh proxy db postgres --db-user postgres --db-name postgres --tunnel
Started authenticated tunnel for the PostgreSQL database "postgres" in cluster "teleport-local" on 127.0.0.1:56158.
To avoid port randomization, you can choose the listening port using the --port flag.

Teleport Connect is a desktop app that can manage database proxies for you.
Learn more at https://goteleport.com/docs/connect-your-client/teleport-connect/#connecting-to-a-database

Use the following command to connect to the database or to the address above using other database GUI/CLI clients:
  $ psql postgres://postgres@localhost:56158/postgres

teleport(master) ✗:
```

tunnel with specific port

```
teleport(master) ✗: tsh proxy db postgres --db-user postgres --db-name postgres --tunnel --port 12345
Started authenticated tunnel for the PostgreSQL database "postgres" in cluster "teleport-local" on 127.0.0.1:12345.

Teleport Connect is a desktop app that can manage database proxies for you.
Learn more at https://goteleport.com/docs/connect-your-client/teleport-connect/#connecting-to-a-database

Use the following command to connect to the database or to the address above using other database GUI/CLI clients:
  $ psql postgres://postgres@localhost:12345/postgres

teleport(master) ✗:
```

regular proxy with random port

```
teleport(master) ✗: tsh proxy db postgres --db-user postgres --db-name postgres
Started DB proxy on 127.0.0.1:56169
To avoid port randomization, you can choose the listening port using the --port flag.

Teleport Connect is a desktop app that can manage database proxies for you.
Learn more at https://goteleport.com/docs/connect-your-client/teleport-connect/#connecting-to-a-database

Use following credentials to connect to the postgres proxy:
  ca_file=/Users/rav/.tsh/keys/teleport-local.dev/cas/teleport-local.pem
  cert_file=/Users/rav/.tsh/keys/teleport-local.dev/rav-db/teleport-local/postgres-x509.pem
  key_file=/Users/rav/.tsh/keys/teleport-local.dev/rav

teleport(master) ✗:
```

regular proxy with specific port

```
teleport(master) ✗: tsh proxy db postgres --db-user postgres --db-name postgres --port 12345
Started DB proxy on 127.0.0.1:12345

Teleport Connect is a desktop app that can manage database proxies for you.
Learn more at https://goteleport.com/docs/connect-your-client/teleport-connect/#connecting-to-a-database

Use following credentials to connect to the postgres proxy:
  ca_file=/Users/rav/.tsh/keys/teleport-local.dev/cas/teleport-local.pem
  cert_file=/Users/rav/.tsh/keys/teleport-local.dev/rav-db/teleport-local/postgres-x509.pem
  key_file=/Users/rav/.tsh/keys/teleport-local.dev/rav

teleport(master) ✗:
```
</details>

### CODE REVIEW

File: tool/tsh/common/proxy.go
Comment (ravicious) [inline tool/tsh/common/proxy.go:1]: Question to db access folks: should I add the ad to AWS proxies as well? It's the first time I hear about them and I'm not sure if Connect is relevant in their context.

**Comment by zmb3** [comment]
I think these should go to stderr, not stdout. People may have existing tooling that parses the standard output of tsh commands, and we don't want to break that with noisy ads.

**Comment by ravicious** [comment]
@zmb3 Is it fine if I print out to stderr in `tsh login` only? Printing those out to stderr in `tsh proxy db` would be rather annoying to implement, unless we decide to add it after the existing output from `tsh proxy db`.

On top of that, I think the change in `tsh proxy db` is much less likely to interrupt any parsing of stdout there. `tsh login` outputs only relevant, somewhat highly structured info to stdout and alerts already go to stderr, so the case for showing the ad in stderr is much stronger imho.

**Review by zmb3 — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

File: tool/tsh/common/proxy.go
Comment (greedy52) [reply tool/tsh/common/proxy.go:1]: If you mean `tsh proxy aws/azure/gcloud`, they for app access. The flow is similar to kube proxy but use a different HTTP middleware for the local proxy.

**Pull request merged**